trick_source/java/src/trick/common/utils/vs/VSDouble.java

```diff
@@ -12,6 +12,23 @@ public VSDouble(double value) {
 
     @Override
     public void fromVariableServer(String string) {
+        // CTLaChance
+        // JUN 07 2017
+        /*
+        Converts the C representation of +- infinity and +- not-a-number to the Java representation
+        so that Double.parseDouble() can correctly assign those values to the variable value.
+        -NaN does not exist in Java and is therefore converted to NaN.
+        */
+        if(string.equals("inf")) {
+            string = "Infinity";
+        }
+        else if(string.equals("-inf")) {
+            string = "-Infinity";
+        }
+        else if(string.equals("-nan") || string.equals("nan")) {
```

Comment:
`Double.parseDouble` is capable of parsing both `"NaN"` and `"-NaN"`. Although we have observed that it returns `NaN` for both, we should not presume it will always do so. All we want to do here is convert the incoming string to something Java can understand. Convert `"nan"` to `"NaN"` and `"-nan"` to `"-NaN"`, and let Java decide what to do with them.